source/server/hot_restarting_child.cc

```diff
@@ -48,7 +48,8 @@ HotRestartingChild::UdpForwardingContext::getListenerForDestination(
 
 HotRestartingChild::HotRestartingChild(int base_id, int restart_epoch,
                                        const std::string& socket_path, mode_t socket_mode)
-    : HotRestartingBase(base_id), restart_epoch_(restart_epoch) {
+    : HotRestartingBase(base_id), restart_epoch_(restart_epoch),
+      parent_terminated_(restart_epoch == 0), parent_drained_(restart_epoch == 0) {
```

Comment:
I had to stare at this for a minute to figure out why it's this way; probably worth a comment `// Initial process doesn't have a parent so it is initialized to drained.` or similar.